src/renderer/caching_shaper.rs

```diff
@@ -48,101 +44,150 @@ struct Asset;
 
 const DEFAULT_FONT_SIZE: f32 = 14.0;
 
-#[derive(new, Clone, Hash, PartialEq, Eq, Debug)]
-struct ShapeKey {
-    pub text: String,
-    pub bold: bool,
-    pub italic: bool,
+#[derive(Clone)]
+pub struct ExtendedFontFamily {
+    pub fonts: Vec<SkriboFont>
```

Comment:
Would it make sense to make this a hashmap with the properties as key?

Reply:
i didn't wanna deviate that much from the original FontFamily type, plus properties are already contained in the FontRef and we can reach them like [this](https://github.com/exoticusd/neovide/blob/9af54bf225cda844ecbd04209e1216c47d453c62/src/renderer/caching_shaper.rs#L66)